File: library/src/main/java/com/jdroid/java/collections/Maps.java
package com.jdroid.java.collections;

import java.util.HashMap;
import java.util.LinkedHashMap;
import java.util.concurrent.ConcurrentHashMap;

public class Maps {
	
	/**
	 * Creates a <i>mutable</i>, empty {@code HashMap} instance.
	 * 
	 * @return a new, empty {@code HashMap}
	 */
	public static <K, V> HashMap<K, V> newHashMap() {
		return new HashMap<>();
	}
	
	/**
	 * Creates a <i>mutable</i>, empty, insertion-ordered {@code LinkedHashMap} instance.
	 * 
	 * @return a new, empty {@code LinkedHashMap}
	 */
	public static <K, V> LinkedHashMap<K, V> newLinkedHashMap() {
		return new LinkedHashMap<>();
	}

	public static <K, V> ConcurrentHashMap<K, V> newConcurrentHashMap() {
		return new ConcurrentHashMap<>();
	}
}

File: jdroid-java-core/src/main/java/com/jdroid/java/collections/Maps.kt
package com.jdroid.java.collections

import java.util.HashMap
import java.util.LinkedHashMap
import java.util.concurrent.ConcurrentHashMap

object Maps {

    /**
     * Creates a *mutable*, empty `HashMap` instance.
     *
     * @return a new, empty `HashMap`
     */
    fun <K, V> newHashMap(): HashMap<K, V> {
        return HashMap()
    }

    /**
     * Creates a *mutable*, empty, insertion-ordered `LinkedHashMap` instance.
     *
     * @return a new, empty `LinkedHashMap`
     */
    fun <K, V> newLinkedHashMap(): LinkedHashMap<K, V> {
        return LinkedHashMap()
    }

    fun <K, V> newConcurrentHashMap(): ConcurrentHashMap<K, V> {
        return ConcurrentHashMap()
    }
}
